### DISCUSSION TITLE
PSA: The default `GLOBAL_APPLY_PERMISSIONS=true` causes Syncthing to re-hash the download dir every startup

### DISCUSSION THREAD

**Discussion by tallzoologytortbatik** [haugene/docker-transmission-openvpn#2829] [General]
## Problem

Having `GLOBAL_APPLY_PERMISSIONS=true` executes a recursive chown on the data dirs on startup of docker-transmission-openvpn. This modifies the ctime of all files in that dir tree. If you have Syncthing replicating this dir with `syncOwnership=true`, it'll consider the changed ctime as a reson to **re-hash** everything under this dir. If this dir happens to be large and to be sitting on rotational media, it could take many hours on every startup.

I don't know the history and implications behind the recursive chown and doing it on every startup by default, so I'm not gonna comment on its rationale. If someone knows better and considers changing it to `false`, that'll eliminate any ctime side effects.

I wrote this this blurb to make the info searchable via Github and other search engines, for others who're banging their heads against the wall with this. 

## Solution

Ensure `GLOBAL_APPLY_PERMISSIONS` is `false` in your setup.

**Comment by pkishino** [comment]
Thanks for the details. I haven’t had a look at that for ages and not sure if I’ll have the time anytime soon so at least I hope this comment will help someone
